sims/verilator/Makefile

```diff
@@ -41,6 +43,37 @@ endif
 default: $(sim)
 debug: $(sim_debug)
 
+#########################################################################################
+# simulaton requirements
+#########################################################################################
+SIM_FILE_REQS += \
+	$(CHIPYARD_RSRCS_DIR)/csrc/emulator.cc \
+	$(ROCKETCHIP_RSRCS_DIR)/csrc/verilator.h \
+
+# the following files are needed for emulator.cc to compile
+SIM_FILE_REQS += \
+	$(TESTCHIP_RSRCS_DIR)/testchipip/csrc/SimSerial.cc \
+	$(TESTCHIP_RSRCS_DIR)/testchipip/csrc/testchip_tsi.cc \
+	$(TESTCHIP_RSRCS_DIR)/testchipip/csrc/testchip_tsi.h \
+	$(TESTCHIP_RSRCS_DIR)/testchipip/csrc/SimDRAM.cc \
+	$(TESTCHIP_RSRCS_DIR)/testchipip/csrc/mm.h \
+	$(TESTCHIP_RSRCS_DIR)/testchipip/csrc/mm.cc \
+	$(TESTCHIP_RSRCS_DIR)/testchipip/csrc/mm_dramsim2.h \
+	$(TESTCHIP_RSRCS_DIR)/testchipip/csrc/mm_dramsim2.cc \
```

Comment:
Why do these have to be copied over using the Make system for verilator, but not VCS?

Reply:
Because these are used directly in `emulator.cc` whereas VCS doesn't use an `emulator.cc`.

Reply:
Actually, for future reference let me clarify more. By default, if you use something like `SimSerial`, the `addResources` will bring in both the `cc` and `v` files. If you don't use it... then those files won't be included. This is the case for VCS... only files necessary for the simulation will be copied over. However, for Verilator, you need to include all `cc/h` files for `emulator.cc` to compile even if you don't necessarily use the corresponding `v` or `scala` source.